samcli/commands/pipeline/init/interactive_init_flow.py

```diff
@@ -0,0 +1,244 @@
+    if location_choice == "2":
+        _generate_from_custom_location()
+    else:
+        _generate_from_app_pipeline_templates()
+
+
+def _generate_from_app_pipeline_templates() -> None:
+    """
+    Prompts the user to choose a pipeline template from SAM predefined set of pipeline templates hosted in the git
+    repository: aws/aws-sam-cli-pipeline-init-templates.git
+    downloads locally, then generates the pipeline config file from the selected pipeline template.
+    """
+    pipeline_templates_local_dir: Path = _clone_app_pipeline_templates()
+
+    pipeline_templates_manifest: PipelineTemplatesManifest = _read_app_pipeline_templates_manifest(
+        pipeline_templates_dir=pipeline_templates_local_dir
+    )
+    # The manifest contains multiple pipeline-templates so select one
+    selected_pipeline_template_manifest: PipelineTemplateManifest = _select_pipeline_template(
+        pipeline_templates_manifest=pipeline_templates_manifest
+    )
+    selected_pipeline_template_dir: Path = pipeline_templates_local_dir.joinpath(
+        selected_pipeline_template_manifest.location
+    )
+    _generate_from_pipeline_template(pipeline_template_dir=selected_pipeline_template_dir)
+
+
+def _generate_from_custom_location() -> None:
+    """
+    Prompts the user for a custom pipeline template location, downloads locally, then generates the pipeline config file
+    """
+    pipeline_template_repo_url: str = click.prompt(text="template Git location")
+    pipeline_template_local_dir: Path = _clone_custom_pipeline_template(repo_url=pipeline_template_repo_url)
+    _generate_from_pipeline_template(pipeline_template_dir=pipeline_template_local_dir)
+    # Unlike app pipeline templates, custom pipeline templates are not shared between different SAM applications
+    # and should be cleaned up from users' machines after generating the pipeline config files.
+    shutil.rmtree(pipeline_template_local_dir, onerror=rmtree_callback)
+
+
+def _generate_from_pipeline_template(pipeline_template_dir: Path) -> None:
+    """
+    Generates a pipeline config file from a given pipeline template local location
+    """
+    pipeline_template: Template = _initialize_pipeline_template(pipeline_template_dir=pipeline_template_dir)
+    context: Dict = pipeline_template.run_interactive_flows()
+    pipeline_template.generate_project(context)
+
+
+def _clone_app_pipeline_templates() -> Path:
+    """
+    clone aws/aws-sam-cli-pipeline-init-templates.git Git repo to the local machine in SAM shared directory.
+    Returns:
+        the local directory path where the repo is cloned.
+    """
+    try:
+        return _clone_pipeline_templates(
+            repo_url=APP_PIPELINE_TEMPLATES_REPO_URL, clone_name=APP_PIPELINE_TEMPLATES_REPO_LOCAL_NAME
+        )
+    except PipelineTemplateCloneException:
+        # If can't clone app pipeline templates, try using an old clone from a previous run if already exist
```

Comment:
Is this logic necessary? Do we really expect it to fail enough times to justify this (potentially fragile, as no guarantee what's in that path) extra logic?

Reply:
this is just to be consistent with the `sam init` experience

Reply:
Consistency is great when a common pattern is made similar so that customers can reuse previous knowledge. This however seems like an edge case with questionable logic, so I see no value that consistency would provide here.

Reply:
The main reason to do **clone** here is to get the up to date pipeline templates (in case if we updated [our pipeline templates](https://github.com/aws/aws-sam-cli-pipeline-init-templates)) other than this, we want to reused those already existing templates (that is why it is in a shared directory).

The pipeline templates are not expected to be actively updated, so this approach looks reasonable, no?

Reply:
Okay so this behavior is useful only if a clone has been done previously, and a new clone fails (e.g. due to no Internet connection).

I'm fine keeping it as-is since we're already doing something similar, but I do think it has negatives:

1. Requires extra care to ensure the target is clean (since we don't have that guarantee)
2. Adds a precedent where we're using both local and global state, which means future developers will have to spend those extra seconds considering why and their caveats